linux/power_operation_script/set_power_script.yml

```diff
@@ -0,0 +1,118 @@
+  ansible.builtin.debug:
+    msg:
+      - "Start test {{ power_script_type }} {{ power_ops }}"
+
+- name: "Set {{ power_script_type }} script for testcase {{ power_script_ops }}"
+  include: set_power_script_{{ power_script_type }}.yml
+  vars:
+    power_ops: "{{ power_ops }}"
+
+- name: "Check the current configuration for testcase {{ power_script_ops }}"
+  ansible.builtin.command: "{{ vmware_toolbox_cmd_path }} script {{ power_ops }} current"
+  register: power_script_result
+  delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Print the current configuration for testcase {{ power_script_ops }}"
+  ansible.builtin.debug: var=power_script_result.stdout
+
+- name: "Check current configuration is expected"
+  ansible.builtin.assert:
+    that:
+      - power_script_result.stdout.strip() == script_file_path
+    fail_msg: "The configuration for {{ power_script_ops }} is {{ power_script_result.stdout }}, not as expected {{ script_file_path }}."
+    success_msg: "The configuration for {{ power_script_ops }} is {{ power_script_result.stdout }}, which is as expected."
+
+# Won't works with untils/reboot.yml for poweroff event
+- name: "Shutdown and then powerOn guest OS"
+  when: power_ops is match('^(poweroff|poweron)')
+  block:
+    - name: "Shutdown VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: "shutdown-guest"
+
+    - name: "Power on VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: "powered-on"
+
+- name: "Suspended guest of target vm and then resumed"
+  when: power_ops is match('^(suspend|resume)')
+  block:
+    - name: "Suspend the VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: 'suspended'
+
+    - name: "Power on VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: "powered-on"
+        force: true
+
+- name: "Update inventory"
+  include_tasks: ../../common/update_inventory.yml
+
+- name: "Get content of file tools.conf"
+  ansible.builtin.command: "cat /etc/vmware-tools/tools.conf"
+  register: tools_conf_content_result
+  delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Set fact of power script config in tools.conf"
+  ansible.builtin.set_fact:
+    power_script_file: |-
+      {%- power_script_type == 'custom' -%}{{ script_file_path.split('/')[-1].strip() }}
+      {%- else -%}{{ script_file_path }}
+      {%- endif -%}
+
+- name: "Check the configuration is expected in tools.conf"
+  ansible.builtin.assert:
+    that:
+      - tools_conf_content_result.stdout is defined
+      - tools_conf_content_result.stdout.find(power_script_file) != -1
+    fail_msg: "Not found the {{ script_file_path }} in tool.conf for {{ power_script_ops }}, which is not as expected"
+    success_msg: "Found the {{ script_file_path }} in tool.conf for {{ power_script_ops }}, which is as expected."
```

Comment:
Checking tools.conf content should be moved after `include: set_power_script_{{ power_script_type }}.yml`